enhancements/network/ovn-hypershift-live-migration.md

```diff
@@ -0,0 +1,303 @@
+
+In KubeVirt's case, the worker nodes will be spun as KubeVirt virtual machines.
+
+There are some network requirements, some of them are imposed by kubevirt 
+live migration and others by kubernetes:
+
+- The IP Kubelet uses to register/communicate should follow the node during live migration.
+- Guest Cluster Network isolation. Guest clusters shouldn’t be able to talk to other guest clusters directly except through public LBs. Guest clusters shouldn’t be able to talk to any infra components except through public LBs.
+- No TCP connection breakage over E/W and N/S during live migration with minimal disruption.
+- Support exposing workers via services. 
+- Allow workers services access to services like kubedns and apiserver from hosted control plane.
+
+To accomplish the goals listed above the ovn-kubernetes default network has to implement some features triggered for an Hypershift KubeVirt VM.
+hypershift kubevirt VM is detected.
+
+## Motivation
+
+To reach GA at hypershift kubevirt provider it has to implement live migration and adding live-migration features to ovn-k default pod network
+is possible without much impact. This will allow users to spin hypershift kubevirt hosted clusters out of the box without setting up 
+the network at a secondary interface.
+
+### User Stories
+
+I, a Hypershift user, need to migrate one hosted cluster without affecting different ones and have minimal
+disruption.
+
+### Goals
+
+Hypershift kubevirt provider needs the live-migration functionality, missing features at CNI need to be implement at the ovn-kuberentes 
+which is the default CNI for openshift, this should work with dual stack.
+
+### Non-Goals
+
+n/a
+
+## Proposal
+
+At kubevirt every VM is executed inside a "virt-launcher" pod; how the "virt-launcher" pod net interface is "bound" to the VM is specified by the network binding at the VM spec.
+
+The main VM network binding mechanism are:
+- masquerade binding: the VM has a stable fake address that is masqueraded (SNAT) at virt-launcher pod. 
+- bridge binding: The virt-launcher interface is replaced by a bridge with one leg on the pod interface and the other at the VM tap device. When IPAM is configured on the pod interface, a DHCP server is started to advertise the pod's IP to DHCP aware VMs.
+
+Since Masquerade binding has a fake address at VM guest it cannot be used for hypershift, the kubernetes nodes should see the address they are using to register to the api server.
+
+The bridge binding is able to expose the pod ip to the VM as expected by hypershift but kubevirt does not support live migration with it, for kubevirt to support live migration with bridge binding 
+at primary interface the OVN kubernetes CNI need to do the following:
+- not set the IP at pod netns
```

Comment:
Maybe we should make it clear here that OVN-Kube is still assigning the IP to the pod, it is just not setting it in the netns? Also should you mention the proposal includes adding the capability to migrate this IP within OVN from one node to another when live migration happens?

Reply:
Added `controller` to the paragraph and the point you mention.